src/visitors.rs

```diff
@@ -12,10 +12,16 @@ use rustc::ty::{Const, Ty, TyCtxt, TyKind, UserTypeAnnotationIndex};
 use rustc::{hir, mir};
 use std::borrow::Borrow;
 use summaries::PersistentSummaryCache;
+use syntax::errors::{Diagnostic, DiagnosticBuilder};
 use syntax_pos;
 
 /// Holds the state for the MIR test visitor.
 pub struct MirVisitor<'a, 'b: 'a, 'tcx: 'b> {
+    /// A place where diagnostic messages can be buffered by the test harness.
```

Comment:
If you end up adding more than just these two callbacks you might want to consider a trait instead.

Reply:
The thought crossed my mind as well, but hopefully this is it and it seems more light weight.